lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -219,6 +223,42 @@ static OptionalParseResult customTypeParser(AsmParser &parser, StringRef name,
 
     return result = BundleType::get(context, elements, isConst), success();
   }
+  if (name.equals("openbundle")) {
+    SmallVector<OpenBundleType::BundleElement, 4> elements;
+
+    auto parseBundleElement = [&]() -> ParseResult {
```

Comment:
You could probably make a `parseBundleElement` function templated on the element type to unify the open and non-bundle bundle parsing, but I don't think it's a big deal.